File: UVA/11340.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	ios::sync_with_stdio(false);
	int t; cin >> t;

	while(t--) {
		map<char, double> cost;

		int chars; cin >> chars;
		while(chars--) {
			char c; double v; cin >> c >> v;
			cost[c] = v;
		}

		int lines; cin >> lines;
		double total = 0;
		cin.ignore(); cin.ignore();
		while(lines--) {
			string tmp; getline(cin, tmp);
			for(auto c : tmp) {
				if(cost.find(c) != cost.end()) {
					total += cost[c];
				}
			}
		}

		printf("%0.2lf$\n", total / 100.0);
	}

	return 0;
}

File: UVA/11340.py
for _ in range(int(input())):
	cost = {}
	for _ in range(int(input())):
		c, v = input().split()
		cost[c] = int(v)
	total = 0
	for _ in range(int(input())):
		for c in input():
			if c in cost:
				total += cost[c]
	print('{0:.2f}$'.format(total / 100))